### ISSUE TITLE
0.12.x Beta: No audio when playing EAC3 on CCwGTv

### ISSUE THREAD

**Issue by arpitgupta** [jellyfin/jellyfin-androidtv#1032]
**Media Info of the file**

```
Audio
Title English - Dolby Digital+ - 6 ch
Language eng
Codec EAC3
Channels 6 ch
Bitrate 768 kbps
Sample rate 48000 Hz
Default No
```

```
Title 1080p HEVC SDR
Codec HEVC
Profile Main 10
Level 120
Resolution 1920x960
Aspect ratio 2:1
Interlaced No
Framerate 23.976025
Bitrate 4777 kbps
Video range SDR
Color space bt709
Color transfer bt709
Color primaries bt709
Pixel format yuv420p10le
Ref frames 1
```
**System (please complete the following information):**
CCwGTv with 0.12 latest beta
JF: 10.7.6

**Additional context**
When i play content with EAC3 format on my CCwGTv there is no audio. It is connected to my TV and there is no recieiver. From JF dashboard looks like the content was direct playing. The audio should have been transcoded. If i change the JF app to downmix audio to stero then it plays fine. Same content plays fine on my nvidia shield which is connected to a reciever. 

FTR other audio formats such as AAC 5.1 play just fine on CCwGTv and are direct playing.

**Comment by nielsvanvelzen** [comment]
Is this issue new in 0.12 or does it also apply to 0.11?

**Comment by arpitgupta** [comment]
I am not sure about that. I was not using JF regularly with 0.11 as everything was getting transcoded and i had to disabel all transcoding settings for the user.

**Comment by newton181** [comment]
I was facing same issue with Android version, till update the Android webview, its webwiew related with exoplayer?

In Android TV use libvlc automatically, but i will try exoplayer, because recently the TV update the webview.

**Comment by arpitgupta** [comment]
> I was facing same issue with Android version, till update the Android webview, its webwiew related with exoplayer?
> 
> In Android TV use libvlc automatically, but i will try exoplayer, because recently the TV update the webview.

Interesting my webview update gets stuck. Do you have the version that is installed on yours? I can see what i am running.

**Comment by arpitgupta** [comment]
fyi i updated to latest android webview 92.0.4515.115 (July 23, 2021) and i still have this issue.

**Comment by johnsheldon** [comment]
I've had issues playing AC3 audio tracks with Exoplayer #1029, could this be related?

**Comment by Cloneg** [comment]
Had same problem. Workaround is to use external app like vlc for videos with eac3 audio codecs.
More info:
https://www.reddit.com/r/jellyfin/comments/kc9f10/jellyfin_android_app_does_unnecessary_transcoding/

**Comment by newton181** [comment]
> 
> 
> fyi i updated to latest android webview 92.0.4515.115 (July 23, 2021) and i still have this issue.
> 
> Some more info.
> 
> When i set the player to exoplayer, it tries to direct play and there is no sound
> When i set the player to libVLC then we have transcoding and the reason being audiocodec not supported. I have confirmed other audio codec's like AAC 5.1 still continue to directplay.

My bad, i was using an user with transcoded audio, in my main user (without transcode permissions) i still have this issue.

**Comment by johnsheldon** [comment]
So there is a setting to enable surround sound for Chromecast with Google TV that I was unaware of, for some reason it is disabled by default. Audio and full Dolby Vision video are now working in Exoplayer for me, not sure if this has anything to do with your issue but it fixed mine.

**Comment by arpitgupta** [comment]
I have surround setting enabled and i still have the issue. When i try to manually enable various surround sound formats it says my device does not support it, so i did not enable them.

**Comment by svemonix** [comment]
I think this is a typical Android TV issue where by default exoplayer assumes that the audio capabilities are the ones from the device running Android TV. This obviously doesn't work if there are some limitations somewhere in the audio chain (e.g. the Android TV can passthrough EAC3 but the TV doesn't handle EAC3). 

This is why players like Kodi or Plex allows you to fine tune the specific codecs allowed manually. I think this should be implemented in Jellyfin too.

That being said, what is the issue in your specific case? It sounds like the CCwGTv is simply connected to a stereo TV so the downmix output is the correct option.

**Comment by arpitgupta** [comment]
> This is why players like Kodi or Plex allows you to fine tune the specific codecs allowed manually. I think this should be implemented in Jellyfin too.

I did not specify any codec's manually in Plex and it direct stream's this file where only the audio is being converted. I am not sure if Plex provides this option. At least i could not find it in the android app.

**Comment by svemonix** [comment]
Exactly, Plex transcodes the audio on the server side because you probably don’t have HDMI passthrough active so it’s gonna output Stereo like Jellyfin with downmix.
